src/nodes.coffee

```diff
@@ -1157,11 +1157,21 @@ exports.Obj = class Obj extends Base
   shouldCache: ->
     not @isAssignable()
 
+  # Check if object contains splat.
+  hasSplat: ->
+    splat = yes for prop in @properties when prop instanceof Splat
+    splat ? no
+
   compileNode: (o) ->
     props = @properties
     if @generated
       for node in props when node instanceof Value
         node.error 'cannot have an implicit value in an implicit object'
+
+    # Object spread properties. https://github.com/tc39/proposal-object-rest-spread/blob/master/Spread.md
+    # `obj2 = {a: 1, obj..., c: 3, d: 4} => obj2 = Object.assign({}, {a: 1}, obj1, {c: 3, d: 4})`
```

Comment:
This comment is duplicated a bit below. I'd say we remove this one and keep the other.